frappe/desk/form/save.py

```diff
@@ -16,9 +16,11 @@ def savedocs(doc, action):
 
 	# action
 	doc.docstatus = {"Save": 0, "Submit": 1, "Update": 1, "Cancel": 2}[action]
-
 	if doc.docstatus == 1:
-		doc.submit()
+		if doc.meta.submit_in_background and doc.meta.is_submittable:
```

Comment:
A problem that could arise from this is, if someone has done

```
x = doc.submit()
```

We return the whole doc object (i think) after submission and in case of queueing we won't get that and it will cause a lot of problems :P

Reply:
hey but I don't think there would be immediate breaking changes, since by default it's 0

Reply:
Submitting from code and submitting from frontend is different right? 

We can expose same API for when someone wants to submit things from code. Doing this in code wouln't make sense.


Imagine a code that expects a submitted document after calling `doc.submit` they wouldn't actually see a submitted document.

Reply:
let's do this via savedocs ..seems waayyy safer